lang/test/org/partiql/lang/eval/evaluatortestframework/AstEvaluatorTestAdapter.kt

```diff
@@ -112,58 +185,80 @@ class AstEvaluatorTestAdapter : EvaluatorTestAdapter {
                 assertEquals(
                     expectedIonResult,
                     actualIonResult,
-                    assertionMessage
-                )
+                    unexpectedResultErrorCode
+                ) { tc.testDetails(actualIonResult.toString()) }
             }
             ExpectedResultFormat.PARTIQL -> {
-                val expected = try {
+                val expectedExprValueResult = assertDoesNotThrow(
+                    EvaluatorTestFailureReason.FAILED_TO_EVALUATE_PARTIQL_EXPECTED_RESULT,
+                    { tc.testDetails(note) }
+                ) {
                     pipeline.compile(expectedResult).eval(session)
-                } catch (e: Throwable) {
-                    showTestCase()
-                    e.printStackTrace()
-                    Assert.fail("Exception while attempting to evaluate the expected value, see standard output $assertionMessage")
-                    throw e
                 }
-                if (!expected.exprEquals(actualResult)) {
-                    showTestCase()
-                    println("Expected : $expected")
-                    println("Actual   : $actualResult")
 
-                    Assert.fail("Expected and actual ExprValue instances are not equivalent $assertionMessage")
+                if (!expectedExprValueResult.exprEquals(actualExprValueResult)) {
+                    throw EvaluatorAssertionFailedError(
+                        EvaluatorTestFailureReason.UNEXPECTED_QUERY_RESULT,
+                        tc.testDetails(actualExprValueResult.toString())
+                    )
                 }
                 Unit
             }
             ExpectedResultFormat.STRING -> {
-                assertEquals(expectedResult, actualResult.toString(), "Actual result must match expected (string equality)")
+                val actualResultString = actualExprValueResult.toString()
+                assertEquals(
+                    expectedResult,
+                    actualResultString,
+                    EvaluatorTestFailureReason.UNEXPECTED_QUERY_RESULT,
+                ) { tc.testDetails(actualResultString) }
```

Comment:
(Similar to L189 and L202) Should `note` be the first argument here in the call to `testDetails`?

Reply:
yes